### PR TITLE
Add ARGON2I support to the hasher

### PR DESCRIPTION
**Pull request by rullzer**
When on php7.2 we can use the new and improved ARGON2I hashing.
This adds support for that to the hasher. When verifying an old hash
we'll update rehash to move all hashes eventually to the new hash
function.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Comment by rullzer** [comment]
Failure is just swift.

Basically this change should be transparent to the users of the class.

File: lib/private/Security/Hasher.php [hunk 1]
```diff
@@ -149,6 +174,8 @@ public function verify(string $message, string $hash, &$newHash = null): bool {
 
 		if(isset($splittedHash['version'])) {
```
Comment (ChristophWurst) [inline lib/private/Security/Hasher.php]: *split :wink:

**Review by ChristophWurst — APPROVED** [review]
Didn't test but code looks good

**Review by MorrisJobke — APPROVED** [review]
Tested and works with PHP 7.0 and the fallback code.

**Pull request merged**